### PR TITLE
Target android sdk 29

### PR DESCRIPTION
**Pull request by imsodin**
This PR bumps our target sdk level to 29 (android 10), requesting legacy storage. This is a requirement for publishing google play updates (caveat at the end). The behaviour *shouldn't* change, but what should and shouldn't happen on android is a mystery, so I'll add a call for testing/feedback on the forum once this is in a release candidate.

@Catfriend1 A sanity check if I missed something obvious would be appreciated. I did follow the Android 11 troubles you had, but I believe those problems should not come up as we still target android 10/sdk 29. 

Google play caveat:  
<strike>Google is blocking us at the moment anyway, as suddenly our coarse access location permission for restricting sync by wifi name seems to be a problem - I wrote to the support, I am expecting a resolution in 6-8 weeks. I intend to keep releasing as usual, users annoyed by not getting those release should switch to F-Droid (users should anyway switch to F-Droid ;) ). </strike>  
Google support has responded, details in https://forum.syncthing.net/t/how-to-react-to-google-play-restrictions/15872/26

For reference:  
 - WiFi permissions: https://developer.android.com/guide/topics/connectivity/wifi-scan
 - Deprecated method: https://developer.android.com/reference/android/net/wifi/WifiManager.html#getConfiguredNetworks()

### CODE REVIEW

**Comment by Catfriend1** [comment]
Hi, thanks for working on it @imsodin . I'll give https://build.syncthing.net/repository/download/SyncthingAndroid_Build/85323:id/apk/debug/app-debug.apk a quick shot on my Android 10 phone.

**Comment by imsodin** [comment]
Fixed location permission handling to be sdk29/android 10 compatible. And the method to get configured wifis has been deprecated, apparently it's too dangerous to let the users decide if they want to share that info with an app. So now we just display already stored wifis plus the currently connected wifi.

This might not build as the docker image was changed and isn't deployed yet - a local docker build worked fine.

**Comment by imsodin** [comment]
@AudriusButkevicius If you have a moment, I'd appreciate a quick review. I mostly just removed code, but there's a few changes/additions and I'd appreciate getting a check if the little code I wrote managed to violated any (java) best practices or whatever :D (no hurry though, I intend to skip a Syncthing release to have this in beta for a while).

**Comment by Catfriend1** [comment]
@imsodin Could I test shot the Apk from teamcity on my phone or is it somehow non representative for testing because of the outdated docker? In case you'd like me to test it, I'd appreciate an APK :).

**Comment by imsodin** [comment]
I changed the sdk tools. Weirdly enough that doesn't seem to be a requirement (CI still builds fine). I don't know how much of a difference it makes. My locally built apk has a separate id  (com.nutomic.syncthingandroid.debug) and name to make it co-installable with the normal release - if you want to test that: https://www.swisstransfer.com/d/9c6f38ca-da71-418b-ac0f-32c1df426786

**Review by Catfriend1 — APPROVED** [review]
Works on my Android 9 phone. If it isn't connected to any WiFi and one hits the button "choose whitelisted WiFi ssids" then it shows "<unknown ssid>". I think that is required for fail-safe... so you did it right and do not end up with an empty (null) list.

**Comment by Catfriend1** [comment]
Did not do a test but looks even better now :)

**Comment by Catfriend1** [comment]
@imsodin Do you need more help or testing on this? If required, I'd take another of your APK builds for a test.

**Comment by imsodin** [comment]
Thanks for looking into it and testing on your device. I also did some basic testing on my android 10 device. I think we are fine on that front :)

I am waiting for a review by @AudriusButkevicius (or a "just go ahead, no time or interest" ;) ).

**Comment by auanasgheps** [comment]
Sorry for spamming but please [consider this issue](https://github.com/syncthing/syncthing-android/pull/1566#issuecomment-737781698) before bumping up target SDK.

**Comment by imsodin** [comment]
This is skd 29 / android 10 - it's not related to the all file permissions thing, we are still requesting legacy storage handling here.

If you want you can test an apk from CI or the link above and see if it still works using a "root path" - it should. Otherwise there will be an beta release with this to test.

**Comment by auanasgheps** [comment]
@imsodin tested the debug version and it works! Thanks!

So this is going to be like this in the future? If yes, sorry for bothering you.

**Comment by imsodin** [comment]
@AudriusButkevicius could you please have a look, now that v1.12.1-rc.1 is here? Thanks!

> I am waiting for a review by @AudriusButkevicius (or a "just go ahead, no time or interest" ;) ).

**Comment by AudriusButkevicius** [comment]
Sorry. Completely missed this. As I though it was just chatter. I'll have a look later today

File: app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java
Comment (AudriusButkevicius) [inline app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java:79]: This lost the null check, and will blow up with NPE?

File: app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java
Comment (AudriusButkevicius) [inline app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java:87]: How do we know it's not a perms flop?
I guess we might not have asked for the permission at this point?

File: app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java
Comment (imsodin) [reply app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java:79]: I don't see what might be null here? We create `selected` ourselves - it may be empty but not null. And `ssid` is a string.

Comment (AudriusButkevicius) [reply app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java:79]: Strings are nullable in java, so ssid.contains will blow up if ssid is null.

Comment (AudriusButkevicius) [reply app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java:79]: The comment before even talks about ithttps://github.com/syncthing/syncthing-android/issues/620 in

Comment (Catfriend1) [reply app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java:79]: Audrius is right, I've had tests in the past I've documented in code where ssid was returned as null by Android. That happens when location is turned off.
See https://github.com/Catfriend1/syncthing-android/blob/d38f56f6967447680020eeaa3828367a4599d653/app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreferenceBase.java#L131

Comment (imsodin) [reply app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java]: Right, right, `String` is a class... Missed the comment because it was part of the code I just discarded due to `getConfiguredNetworks` deprecation. At least I know now why I asked for a review :|  
Thanks!

**Comment by imsodin** [comment]
Good to merge now?

**Review by AudriusButkevicius — APPROVED** [review]

**Pull request merged**
